fedimint-client/src/db.rs

```diff
@@ -262,3 +278,250 @@ impl_db_lookup!(
     key = ClientModuleRecovery,
     query_prefix = ClientModuleRecoveryPrefix
 );
+
+/// `ClientMigrationFn` is a function that modules can implement to "migrate"
+/// the database to the next database version.
+pub type ClientMigrationFn = for<'r, 'tx> fn(
+    &'r mut DatabaseTransaction<'tx>,
+    ModuleInstanceId,
+    Vec<(Vec<u8>, OperationId)>, // active states
+    Vec<(Vec<u8>, OperationId)>, // inactive states
+    ModuleDecoderRegistry,
```

Comment:
Do we actually need this? The module should be able to decode its own SMs and they shouldn't contain any data specific to other modules? Maybe with the exception of the transaction submission SM, which doesn't belong to a module anyway though. Let's keep it for flexibility though for now.

Reply:
I don't think its strictly necessary but I included it because I wanted to use `DynState::from_bytes` to convert to the typed states. I'm ok with keeping it for flexibility.